sync/class.jetpack-sync-module-posts.php

```diff
@@ -169,9 +169,9 @@ function expand_wp_insert_post( $args ) {
 		$post         = $args[1];
 		$update       = $args[2];
 		$is_auto_save = isset( $args[3] ) ? $args[3] : false; //See https://github.com/Automattic/jetpack/issues/7372
-		$just_published = isset( $args[4] ) ? $args[4] : false; //Preventative in light of above issue
+		$previous_state = isset( $args[4] ) ? $args[4] : false; //Preventative in light of above issue
 
-		return array( $post_id, $this->filter_post_content_and_add_links( $post ), $update, $is_auto_save, $just_published );
+		return array( $post_id, $this->filter_post_content_and_add_links( $post ), $update, $is_auto_save, $previous_state );
```

Comment:
Should we consider simply sending a $state parameter that is an array with keys 'is_autosave', 'previous_state', etc.?

Reply:
I didn't want to change things up to much. In terms of what the data shape is that we pass back. But now is the time to do that. I think we could definitely do that.